lib/debt_management_center/financial_status_report_downloader.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'net/http'
+require 'uri'
+
+module DebtManagementCenter
+  class FinancialStatusReportDownloader
+    def initialize(financial_status_report)
+      @financial_status_report = financial_status_report
+    end
+
+    def download_pdf
+      uri = URI.parse(
+        "#{Settings.dmc.url}financial-status-report/documentstream?objectId=#{@financial_status_report.filenet_id}"
+      )
+      http = Net::HTTP.new(uri.host, uri.port)
+      request = Net::HTTP::Get.new(uri.request_uri)
```

Comment:
Instead of using Net::Http could the Common::Client service pattern be used here along with the existing perform method?

Reply:
Yes, I initially tried to go that route but it seemed to be more trouble than it's worth, as it's designed around accepting JSON responses by default, and there wasn't a good class between the SOAP and REST for handling this particular use case, where the response is coming back as binary data which in turn gets presented to the user as a PDF.

Reply:
Ah, ok. Your approach makes sense then!